course/models.py

```diff
@@ -122,17 +122,33 @@ class Meta:
 
 class ReviewManager(models.Manager):
 
-    def get_reviews_by_course(self, course: int):
+    def get_reviews_by_course(self, course: TeacherCourse):
         reviews = Review.objects.filter(course=course)
         return reviews
 
-    def get_avg_rating_by_course(self, course: int):
+    def get_avg_rating_by_course(self, course: TeacherCourse):
         avg_rating = self.filter(course=course).aggregate(models.Avg('rating'))['rating__avg']
         return avg_rating
 
-    def get_number_of_review_of_course(self, course):
+    def get_number_of_review_of_course(self, course: TeacherCourse):
         return len(self.filter(course=course))
 
+    def check_if_review_exist_by_student(self, student: User, course: TeacherCourse):
+        reviews_by_course = self.get_reviews_by_course(course)
+        try:
+            return reviews_by_course.get(student=student)
+        except Review.DoesNotExist:
+            return None
+
+    def check_if_review_made_by_student(self, student: User, course: TeacherCourse):
```

Comment:
Where are you using this method?

Reply:
In the end I didn't use it, I will delete it

Reply:
Done